writer.go

```diff
@@ -240,6 +253,7 @@ func (w *Writer) WriteMessages(ctx context.Context, msgs ...Message) error {
 		timer.Stop()
 
 		if err != nil {
+			err = fmt.Errorf("failed to write message: %s", err)
```

Comment:
What is the purpose of overriding this error? It seems like it would discard important type information about the problem (for example if the original error is a `*net.OpError` or `context.DeadlineExceeded` it coudl be introspected by the program).

Reply:
good point!